mtags/src/main/scala/scala/meta/internal/mtags/CommonMtagsEnrichments.scala

```diff
@@ -506,7 +506,12 @@ trait CommonMtagsEnrichments {
 
     def isScalaOrJava: Boolean = {
       toLanguage match {
-        case Language.SCALA | Language.JAVA => true
+        case Language.SCALA | Language.JAVA =>
+          /* if the file doesn't yet exist we can only guess if it will be a file
+           * we assume if it has a right extension that it's a file.
+           */
+          !Files.exists(path.toNIO) ||
+          Files.isRegularFile(path.toNIO)
```

Comment:
wdyt about extracting this to function with scaladoc? `Files.exists(path.toNIO)` is important, but I'd emphasise  why it's needed - because `Files.isRegularFile` returns false for non-exsiting files

Reply:
I added the scaladoc to the entire method, since the two lines don't make much sense standalone.